backends/gstreamer/src/media_stream_source.rs

```diff
@@ -0,0 +1,211 @@
+use glib;
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use gst;
+use gst::prelude::*;
+use gst::subclass::prelude::*;
+use media_stream::GStreamerMediaStream;
+use url::Url;
+
+// Implementation sub-module of the GObject
+mod imp {
+    use super::*;
+
+    pub struct ServoMediaStreamSrc {
+        cat: gst::DebugCategory,
+        proxysrc: gst::Element,
+        srcpad: gst::GhostPad,
+    }
+
+    impl ServoMediaStreamSrc {
+        pub fn set_stream(&self, stream: &mut GStreamerMediaStream) {
+            gst_log!(self.cat, "Setting stream");
+
+            // Append a proxysink to the media stream pipeline.
+            let pipeline = stream.pipeline_or_new();
```

Comment:
Can we just use `attach_to_pipeline` here? Is there a reason there needs to be a proxysrc/sink? The GstreamerMediaStream may already arise from a proxysrc element.

Reply:
What's the benefit of using `attach_to_pipeline` over the current approach?

In any case, I am not sure if this is possible. The GstPlayer [pipeline's](https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-bad-libs/html/GstPlayer.html#gst-player-get-pipeline) (playbin) source is [read-only](https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-base-plugins/html/gst-plugins-base-plugins-playbin.html#GstPlayBin--source), so I am not sure if I can modify the pipeline directly. I can try if you think it is a better option though.

Reply:
> What's the benefit of using attach_to_pipeline over the current approach?

It avoids an extra proxy.

You don't add it to the playbin's pipeline, you add it to `ServoMediaStreamSrc::constructed`. May need a new version of `attach_to_pipeline` for `attach_to_bin`.

Right now the code adds a proxysrc to `ServoMediaStreamSrc`s bin, instead we should attach the entire `MediaStream` pipeline to it.

This doesn't have to be done in this PR, and I can try to do it later if you want.

Reply:
I guess setting up the ghost pads is trickier if you want to do this, since you don't have the streams during construction. Hmm.

Reply:
Ok, I see.

I think that should be fine as long as we set the ghost pad before setting the player's uri.

Reply:
Oh, as in we just don't connect the ghost pad to anything? That could work, I'm not sure if ghost pads can do that.

Reply:
We need to connect the ghost pad to something and I thought that that something could be a `fakesrc` or such. But it does not seem to work :\

Reply:
Yeah. ANyway, we don't have to solve this yet.